modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -208,9 +224,20 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
     }
 
     this.urls = urls;
+    this.exclusionPattern = exclusionPattern;
+  }
+
+  private boolean isExcluded(String url) {
+    boolean isExcluded = false;
+    Pattern exclusionPattern = this.exclusionPattern;
+    if (exclusionPattern != null) {
+      Matcher matcher = exclusionPattern.matcher(url);
```

Comment:
We use the `matcher` just once, so we might as well just write `exclusionPattern.matches(url)`

This method could basically be

```
return exlcusionPattern != null && exclusionPattern.matches(url);
```

But I could understand if people find that too terse. :wink:

Reply:
> We use the matcher just once, so we might as well just write exclusionPattern.matches(url)

No, we use a pre-compiled regex (`Pattern`)  which is not supported by the method `matches`.

`return exlcusionPattern != null && exclusionPattern.matches(url);`

No, that code is not thread-safe.

Reply:
Nevermind, I misread the docs. It's too hot. :weary: